### PR TITLE
CEDS-1215: created notification service and added tests

### PR DESCRIPTION
**Pull request by PatrykRudnicki**
Additionally changed Retrievals to use not deprecated and the same with body parsers.

### CODE REVIEW

File: app/uk/gov/hmrc/exports/services/NotificationService.scala [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        metrics.incrementCounter(notificationMetric)
+        logger.error("There was a problem during saving notification")
+        Accepted
+      }
+  }
+
+  private def buildStatus(responses: Seq[Response]): Option[String] =
+    responses.map { response =>
+      (response.functionCode, response.status.flatMap(_.nameCode).headOption) match {
+        case ("11", Some(nameCode)) if nameCode == "39" || nameCode == "41" => s"11$nameCode"
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/services/NotificationService.scala]: MAGIC NUMBERS !

File: app/uk/gov/hmrc/exports/services/NotificationService.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  private val logger = Logger(this.getClass)
+
+  def save(notification: DeclarationNotification): Future[Result] = {
+    val eori = notification.eori
+    val convId = notification.conversationId
+
+    for {
+      oldNotification <- notificationsRepository
+        .getByEoriAndConversationId(eori, convId)
+        .map(_.sortWith((a, b) => a.isOlderThan(b)).headOption)
```
Comment (MaciejRewera) [inline app/uk/gov/hmrc/exports/services/NotificationService.scala]: I am not entirely sure, but isn't it that now we will have the oldest notification at the head? I thought we want to get the "youngest" notification stored so far.

File: test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import uk.gov.hmrc.exports.models.{DeclarationMetadata, DeclarationNotification}
+import uk.gov.hmrc.exports.repositories.{NotificationsRepository, SubmissionRepository}
+import uk.gov.hmrc.exports.services.NotificationService
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.wco.dec.{Response, ResponseStatus}
+import util.NotificationTestData
+
+import scala.concurrent.Future
+
+class NotificationsServiceSpec
```
Comment (MaciejRewera) [inline test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala]: Can we add a test where we have several (2-3) notifications stored already and we receive another one?

File: test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      testScenario(notificationToSave, notificationSaveResult = true, submissionUpdateResult = true)
+    }
+
+    "return ACCEPTED when notification is not able to be persisted" in new SetUp {
+      val notificationToSave: DeclarationNotification =
+        submissionNotification.copy(dateTimeReceived = submissionNotification.dateTimeReceived.minusHours(2))
+      testScenario(notificationToSave, notificationSaveResult = false, submissionUpdateResult = true)
+    }
+
+    "test buildStatus maps 39 correctly to CustomsPositionGranted when function code is 11" in new SetUp {
```
[the remark this answers is not in the document]
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala]: I put those values inside vals

File: app/uk/gov/hmrc/exports/services/NotificationService.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/services/NotificationService.scala]: MAGIC STRINGS! not numbers 😂 

Fixed! :)

File: app/uk/gov/hmrc/exports/services/NotificationService.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/services/NotificationService.scala]: This method is covered by tests.

`DeclarationNotificationSpec`. I'll check that it covers all logic, but the method should works correctly

Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/services/NotificationService.scala]: added filtering by date and improved `isOlderThan` method

File: test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/services/NotificationsServiceSpec.scala]: of course, two cases added

**Pull request merged**